router/core/transport.go

```diff
@@ -122,6 +127,19 @@ func (t TransportFactory) RoundTripper(transport http.RoundTripper, enableStream
 					}
 					span.SetAttributes(otel.WgOperationType.String(operation.opType))
 				}
+
+				var subgraph *nodev1.Subgraph
```

Comment:
We should create a separate struct for subgraphs and embed them in the RequestContext e.g. `subgraphs()` (read-only).
Additionally, for convenience, we should expose a method like `activeSubgraph()` to get the current subgraph based on the subgraph request. This could return `nil` when evaluated at an early stage like [`middleware`](https://github.com/wundergraph/cosmo/blob/main/router/core/modules.go#L56). The data can be checked [here](https://github.com/wundergraph/cosmo/blob/main/router/core/transport.go#L41).

That also means you could replace the code with:

```go
rc := getRequestContext(r.Context())
if s:= rc.activeSubgraph(); s != nil {
	// .....
}
```

Reply:
Done